examples/client/qt5client.cpp

```diff
@@ -0,0 +1,133 @@
+/*
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#include <QApplication>
+#include <iostream>
+#include <thread>
+#include <QPaintEvent>
+#include <QPainter>
+#include "rfb/rfbclient.h"
+#include <QWidget>
+
+class VncViewer : public QWidget
+{
+    //if you want to make use of signals/slots, uncomment the line bellow:
+    //Q_OBJECT
+public:
+    VncViewer(QWidget *parent = nullptr) {};
+    virtual ~VncViewer() {};
+    void start();
+    std::string serverIp;
+    int serverPort;
+    std::thread *vncThread() const;
+    void paintEvent(QPaintEvent *event) override;
+
+private:
+    QImage m_image;
+    rfbClient *cl;
+    std::thread *m_vncThread;
+    static void finishedFramebufferUpdateStatic(rfbClient *cl);
+    void finishedFramebufferUpdate(rfbClient *cl);
+};
+
+void VncViewer::finishedFramebufferUpdateStatic(rfbClient *cl)
+{
+    VncViewer *ptr = (VncViewer *) rfbClientGetClientData(cl, nullptr);
```

Comment:
auto ptr =static_cast<VncViewer>(rfbClientGetClientData(cl, nullptr));

Reply:
Done at [4137cab](https://github.com/LibVNC/libvncserver/pull/612/commits/4137cab13e56cc9e186f7850e59beac780206918). I didn't use auto because I wanted to keep the type explicit, since this is an example.

Reply:
It's explicit at right hand AAA